### PR TITLE
[Impeller] Fix uniform buffer bindings to point to the right var

### PR DESCRIPTION
**Pull request by iskakaushik**

### CODE REVIEW

File: DEPS [hunk 1]
```diff
@@ -18,7 +18,7 @@ vars = {
   'llvm_git': 'https://llvm.googlesource.com',
   # OCMock is for testing only so there is no google clone
   'ocmock_git': 'https://github.com/erikdoe/ocmock.git',
-  'skia_revision': '14ebc89de8b8ec7d350b3743ad4c14a969e4f596',
+  'skia_revision': 'e89e47d02a2338c72aa09a153770cba70571ad99',
```
Comment (chinmaygarde) [inline DEPS:21]: Did you mean to stage this?

Comment (iskakaushik) [reply DEPS:21]: nope

File: impeller/fixtures/sample.vert [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 #include "types.h"
 
-uniform UniformBufferObject {
+layout (std140, binding = 17) uniform UniformBufferObject {
```
Comment (chinmaygarde) [inline impeller/fixtures/sample.vert:7]: Perhaps specify the set as well?

Comment (chinmaygarde) [reply impeller/fixtures/sample.vert:7]: Since the default is zero, it may be easy to miss.

File: impeller/compiler/compiler_unittests.cc [hunk 3]
```diff
@@ -90,6 +90,41 @@ TEST_P(CompilerTest, BindingBaseForFragShader) {
...
 }
 
+TEST_P(CompilerTest, UniformsHaveBindingAndSet) {
+  if (GetParam() == TargetPlatform::kFlutterSPIRV) {
+    // This is a failure of reflection which this target doesn't perform.
+    GTEST_SKIP();
+  }
+
+#ifndef IMPELLER_ENABLE_VULKAN
+  GTEST_SKIP();
```
Comment (chinmaygarde) [inline impeller/compiler/compiler_unittests.cc:100]: This should work on all backends. Specifying the binding and set should not be a Vulkan only thing.

File: impeller/fixtures/sample.vert (hunk 2 above)
Comment (iskakaushik) [reply impeller/fixtures/sample.vert]: done!

File: impeller/compiler/compiler_unittests.cc (hunk 3 above)
Comment (iskakaushik) [reply impeller/compiler/compiler_unittests.cc]: good catch!

**Pull request merged**